src/meta/src/hummock/compaction_scheduler.rs

```diff
@@ -161,14 +161,13 @@ where
                     let current_compactor_tasks =
                         self.hummock_manager.list_assigned_tasks_number().await;
                     tracing::warn!("No idle compactor available. The assigned task number for every compactor is (context_id, count):\n {:?}", current_compactor_tasks);
-                    compact_task.task_status = false;
                     tokio::time::sleep(Duration::from_secs(
                         self.compactor_selection_retry_interval_sec,
                     ))
                     .await;
                     match self
                         .hummock_manager
-                        .cancel_compact_task(&compact_task)
+                        .cancel_compact_task(&mut compact_task)
```

Comment:
Since the task has not been assigned, isn't this a no op?

Reply:
No.
Even it's not assigned, the picked task has held some SSTs. So if not cancelled, they cannot be picked by other task.

Reply:
But, what I'm saying that the actual cancel_compact_task is a noop as we return early if we cannot find task_id in compact_task_assignment

Reply:
I think this is actually a mistake, we should somehow still commit the task cancellation status even if we can't find the task assignment.

Reply:
You're right. Fixed.

Reply:
I think the semantics we want is to verify task assignment id == context_id as an Option. So if None is provided, we expect None. Behaviour could be to Err as long as they do not agree as an option. Then passing context_id or assign_context_id is equivalent, and we do a commit regardless. 

Alternately, if we pass Some context id, we may not Err if task assignment is None

Reply:
Ok, I guess semantics now is, if we pass context_id, we already know we have committed an assignment, so if we can't find task assignment, then it must have been reported already.

Btw, we don't really have correct abort logic for certain things like removing the task assignment. So if anything further on in code block fails, we never revert the removal of the task...

Reply:
Here is current fix:
- When context_id is None, it means we want to succeed anyway, no matter what assignment id is.
- When context_id is Some, it means we want to succeed only when context_id is valid.
  -  If assignment_id != context_id, we return early because context_id shouldn't modify task owned by assignment_id.
- Always use context_id in commit_multi_var, because that's the caller. The purpose of validating context_id in commit_multi_var is to ensure only valid caller can write hummock.

Reply:
> Ok, I guess semantics now is, if we pass context_id, we already know we have committed an assignment, so if we can't find task assignment, then it must have been reported already.

Yes.

> Btw, we don't really have correct abort logic for certain things like removing the task assignment. So if anything further on in code block fails, we never revert the removal of the task...

All changes are committed by `commit_multi_var!(self, context_id, compact_status, compact_task_assignment)?;`.
So if any further on block fails, they are all reverted (supported by commit_multi_var).